pkg/controller/jobs/job/job_multikueue_adapter.go

```diff
@@ -38,12 +38,33 @@ import (
 	"sigs.k8s.io/kueue/pkg/features"
 	"sigs.k8s.io/kueue/pkg/util/api"
 	clientutil "sigs.k8s.io/kueue/pkg/util/client"
+	"sigs.k8s.io/kueue/pkg/workloadslicing"
 )
 
 type multiKueueAdapter struct{}
 
 var _ jobframework.MultiKueueAdapter = (*multiKueueAdapter)(nil)
 
+// updateLabel helper sets the provided label key/value on the provided job.
+// Returns true, if the job's label was changed.
+func updateLabel(job *batchv1.Job, labelKey, labelValue string) bool {
+	labels := job.Labels
+	if labels == nil {
+		job.Labels = map[string]string{labelKey: labelValue}
+		return true
+	}
+	value, found := labels[labelKey]
+	labels[labelKey] = labelValue
+	return !found || value != labelValue
+}
+
+// updateParallelism helper sets the provided parallelism on the provided Job (spec).
+// Returns true, if the job's parallelism value was changed.
+func updateParallelism(job *batchv1.Job, parallelism *int32) bool {
+	job.Spec.Parallelism, parallelism = parallelism, job.Spec.Parallelism
+	return ptr.Deref(job.Spec.Parallelism, 0) != ptr.Deref(parallelism, 0)
+}
```

Comment:
This is called once, then this function does not seem to have much responsibility.
So, I would postpone having such util function.
I proposed how to update this in the following comment.

Reply:
Answered in https://github.com/kubernetes-sigs/kueue/pull/6445#discussion_r2302640674